aws_batch/pipeline.py

```diff
@@ -798,23 +815,25 @@ def run_star(sample_name, fastq_file_1, fastq_file_2, star_genome_s3_path,
         execute_command("aws s3 cp %s %s/" % (star_genome_s3_path, REF_DIR))
         execute_command("cd %s; tar xvfz %s" % (REF_DIR, genome_file))
         logging.getLogger().info("downloaded index")
-    star_command_params = ['cd', scratch_dir, ';', STAR,
-                           '--outFilterMultimapNmax', '99999',
-                           '--outFilterScoreMinOverLread', '0.5',
-                           '--outFilterMatchNminOverLread', '0.5',
-                           '--outReadsUnmapped', 'Fastx',
-                           '--outFilterMismatchNmax', '999',
-                           '--outSAMmode', 'None',
-                           '--clip3pNbases', '0',
-                           '--readFilesCommand', 'zcat',
-                           '--runThreadN', str(multiprocessing.cpu_count()),
-                           '--genomeDir', REF_DIR + '/STAR_genome',
-                           '--readFilesIn', fastq_file_1, fastq_file_2]
-    execute_command(" ".join(star_command_params))
-    logging.getLogger().info("finished job")
-    # extract out unmapped files
-    execute_command("cp %s/%s %s/%s;" % (scratch_dir, 'Unmapped.out.mate1', result_dir, STAR_OUT1))
-    execute_command("cp %s/%s %s/%s;" % (scratch_dir, 'Unmapped.out.mate2', result_dir, STAR_OUT2))
+    # Check if parts.txt file exists, if so use the new version of (partitioned indices). Otherwise, stay put
+    if os.path.isfile("%s/STAR_genome/parts.txt" % REF_DIR):
+        num_parts = int(open("%s/STAR_genome/parts.txt" % REF_DIR, 'rb').read())
+        part_idx = 0
+        tmp_result_dir = "%s/star-part-%d" % (scratch_dir, part_idx)
+        run_star_part(tmp_result_dir, REF_DIR + "/STAR_genome/part-%d" % part_idx, fastq_file_1, fastq_file_2)
+        for i in range(1, num_parts):
+            fastq_1 = "%s/Unmapped.out.mate1" % tmp_result_dir
+            fastq_2 = "%s/Unmapped.out.mate2" % tmp_result_dir
+            tmp_result_dir = "%s/star-part-%d" % (scratch_dir, i)
+            run_star_part(tmp_result_dir, REF_DIR + "/STAR_genome/part-%d" % i, fastq_1, fastq_2)
+        # extract out unmapped files
```

Comment:
nice...

Reply:
nice optimization

should we delete the unmapped files after we are done with them?

curious if for the very big jobs (4GB input) this will increase time due to writing out all the unmapped files... if we split the input into N parts, then the amount of intermediate output will be input_size*(N-1)/2

also for the case when we run this in parallel, may be useful to just turn off the unmapped output?  or send it to /dev/null

Reply:
oh i see we actually need the unmapped ones for the overall result

Reply:
yes